### PR TITLE
importer-rest-api-specs: process nested fields from `allOf` models

### LINKED ISSUE #1714: `SecurityInsights@2022-07-01-preview` - missing properites
generated `MLBehaviorAnalyticsAlertRuleTemplateProperties` missing all of `AlertRuleTemplateWithMitreProperties`

resource-manager/securityinsights/2022-07-01-preview/alertruletemplates/model_mlbehavioranalyticsalertruletemplateproperties.go:6

### PR DESCRIPTION
**Pull request by jackofallops**
fixes #1714

```csharp
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using Pandora.Definitions.Attributes;
using Pandora.Definitions.Attributes.Validation;
using Pandora.Definitions.CustomTypes;


// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See NOTICE.txt in the project root for license information.


namespace Pandora.Definitions.ResourceManager.SecurityInsights.v2021_09_01_preview.AlertRuleTemplates;


internal class MLBehaviorAnalyticsAlertRuleTemplatePropertiesModel
{
    [JsonPropertyName("alertRulesCreatedByTemplateCount")]
    [Required]
    public int AlertRulesCreatedByTemplateCount { get; set; }

    [DateFormat(DateFormatAttribute.DateFormat.RFC3339)]
    [JsonPropertyName("createdDateUTC")]
    public DateTime? CreatedDateUTC { get; set; }

    [JsonPropertyName("description")]
    [Required]
    public string Description { get; set; }

    [JsonPropertyName("displayName")]
    [Required]
    public string DisplayName { get; set; }

    [DateFormat(DateFormatAttribute.DateFormat.RFC3339)]
    [JsonPropertyName("lastUpdatedDateUTC")]
    public DateTime? LastUpdatedDateUTC { get; set; }

    [JsonPropertyName("requiredDataConnectors")]
    public List<AlertRuleTemplateDataSourceModel>? RequiredDataConnectors { get; set; }

    [JsonPropertyName("severity")]
    [Required]
    public AlertSeverityConstant Severity { get; set; }

    [JsonPropertyName("status")]
    [Required]
    public TemplateStatusConstant Status { get; set; }

    [JsonPropertyName("tactics")]
    public List<AttackTacticConstant>? Tactics { get; set; }
}

```

```go
package alertruletemplates

import (
	"time"

	"github.com/hashicorp/go-azure-helpers/lang/dates"
)

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See NOTICE.txt in the project root for license information.

type MLBehaviorAnalyticsAlertRuleTemplateProperties struct {
	AlertRulesCreatedByTemplateCount int64                          `json:"alertRulesCreatedByTemplateCount"`
	CreatedDateUTC                   *string                        `json:"createdDateUTC,omitempty"`
	Description                      string                         `json:"description"`
	DisplayName                      string                         `json:"displayName"`
	LastUpdatedDateUTC               *string                        `json:"lastUpdatedDateUTC,omitempty"`
	RequiredDataConnectors           *[]AlertRuleTemplateDataSource `json:"requiredDataConnectors,omitempty"`
	Severity                         AlertSeverity                  `json:"severity"`
	Status                           TemplateStatus                 `json:"status"`
	Tactics                          *[]AttackTactic                `json:"tactics,omitempty"`
}

func (o *MLBehaviorAnalyticsAlertRuleTemplateProperties) GetCreatedDateUTCAsTime() (*time.Time, error) {
	if o.CreatedDateUTC == nil {
		return nil, nil
	}
	return dates.ParseAsFormat(o.CreatedDateUTC, "2006-01-02T15:04:05Z07:00")
}

func (o *MLBehaviorAnalyticsAlertRuleTemplateProperties) SetCreatedDateUTCAsTime(input time.Time) {
	formatted := input.Format("2006-01-02T15:04:05Z07:00")
	o.CreatedDateUTC = &formatted
}

func (o *MLBehaviorAnalyticsAlertRuleTemplateProperties) GetLastUpdatedDateUTCAsTime() (*time.Time, error) {
	if o.LastUpdatedDateUTC == nil {
		return nil, nil
	}
	return dates.ParseAsFormat(o.LastUpdatedDateUTC, "2006-01-02T15:04:05Z07:00")
}

func (o *MLBehaviorAnalyticsAlertRuleTemplateProperties) SetLastUpdatedDateUTCAsTime(input time.Time) {
	formatted := input.Format("2006-01-02T15:04:05Z07:00")
	o.LastUpdatedDateUTC = &formatted
}
```

### CODE REVIEW

File: tools/importer-rest-api-specs/components/parser/models.go [hunk 1]
```diff
@@ -177,6 +177,34 @@ func (d *SwaggerDefinition) detailsForField(modelName string, propertyName strin
 			}
 			nestedFields[propName] = *nestedField
 		}
+		for _, inlinedModel := range value.AllOf {
+			remotePath := inlinedModel.Ref.String()
+			remotePathParts := strings.Split(remotePath, "/")
+			remoteRef := remotePathParts[len(remotePathParts)-1]
```
Comment (tombuildsstuff) [inline tools/importer-rest-api-specs/components/parser/models.go:183]: heads up we have `fragmentNameFromString` for this

File: tools/importer-rest-api-specs/components/parser/models.go [hunk 2]
```diff
@@ -177,6 +177,34 @@ func (d *SwaggerDefinition) detailsForField(modelName string, propertyName strin
 			}
 			nestedFields[propName] = *nestedField
 		}
+		for _, inlinedModel := range value.AllOf {
```
Comment (tombuildsstuff) [inline tools/importer-rest-api-specs/components/parser/models.go:180]: can we add an example Swagger/test for this?

**Review by tombuildsstuff — APPROVED** [review]
A couple of minor comments but this otherwise LGTM 👍

File: tools/importer-rest-api-specs/components/parser/models.go (hunk 2 above)
Comment (jackofallops) [reply tools/importer-rest-api-specs/components/parser/models.go:180]: Done 👍

**Review by tombuildsstuff — APPROVED** [review]

**Pull request merged**